### ISSUE TITLE
[BUG] GetRealtimeConversationClient fails with ApiKeyCredential

### ISSUE THREAD

**Issue by tom-b-iodigital** [Azure/azure-sdk-for-net#47535]
### Library name and version

Azure.AI.OpenAI 2.1.0

### Describe the bug

Starting from version 2.1.0 the following code will fail:

```
var openAiClient = new AzureOpenAIClient(new Uri("https://xxxxx.openai.azure.com/"), 
    new System.ClientModel.ApiKeyCredential("xxxxx"));
var realtimeClient = openAiClient.GetRealtimeConversationClient("gpt-4o-realtime-preview");
```

By throwing an exception with message: Value cannot be null. (Parameter 'credential')

In version 2.1.0-beta2 this still worked.

### Expected behavior

A realtime conversaton client should be available after the function call without throwing an exception.

### Actual behavior

Exception with message 'Value cannot be null. (Parameter 'credential')' is thrown

### Reproduction Steps

Add following nuget packages to you project:

<PackageReference Include="Azure.AI.OpenAI" Version="2.1.0" />
<PackageReference Include="OpenAI" Version="2.1.0" />

Use the code below to reproduce (replace the x's with actual values):

```
var openAiClient = new AzureOpenAIClient(new Uri("https://xxxxx.openai.azure.com/"), 
    new System.ClientModel.ApiKeyCredential("xxxxx"));
var realtimeClient = openAiClient.GetRealtimeConversationClient("gpt-4o-realtime-preview");
```

Replace the nuget packages to observe the correct behavior

<PackageReference Include="Azure.AI.OpenAI" Version="2.1.0-beta.2" />
<PackageReference Include="OpenAI" Version="2.1.0" />

**Comment by code-Miracle49** [comment]
@tom-b-iodigital Any workaround for this?

**Comment by code-Miracle49** [comment]
Oh okay, thank you  @tom-b-iodigital.

> > [@tom-b-iodigital](https://github.com/tom-b-iodigital) Any workaround for this?
> 
> I just use the beta 2 version of the Azure.AI.OpenAI and the 2.1.0 version of the openai package. That seems to work fine.
> 
> `<PackageReference Include="Azure.AI.OpenAI" Version="2.1.0-beta.2" /> <PackageReference Include="OpenAI" Version="2.1.0" />`

**Comment by octupul** [comment]
They have removed the azure OpenAI credentialing overrides for the 2.1.0 release version when using realtime which forces it to try to fallback to the regular OpenAI endpoints which aren't setup when you initialize in this manner.  Is this being left out by intention for the team because it is a final release or was this a mistake?

**Comment by onionhammer** [comment]
So this is still completely borked now? _tokenCredential is completely ignored and keyCredential is used exclusively now? Where are the regression tests?

**Comment by IEvangelist** [comment]
Yes, this is certainly still a bug that needs to be addressed. When you try targeting NuGet package `2.1.0` of the `Azure.AI.OpenAI`, you cannot successfully call `GetRealtimeConversationClient` if you're attempting to use the `AzureOpenAIClient` constructor that accepts the `Uri` and `ApiKeyCredential`. In my case, using the:

```cs
var client = new AzureOpenAIClient(endpoint, new AzureKeyCredential(key));

// This throws
var conversationClient = client.GetRealtimeConversationClient("gpt-4o-realtime-preview");
```

The exception is:

```
Exception with message 'Value cannot be null. (Parameter 'credential')' is thrown
```

I experienced this too, the workaround as suggested by @tom-b-iodigital does work:

```xml
<PackageReference Include="Azure.AI.OpenAI" Version="2.1.0-beta.2" />
<PackageReference Include="OpenAI" Version="2.1.0" />
```

Hopefully this is fixed in the next release. /cc @trrwilson FYI.

**Comment by goosenn** [comment]
> Any plans for fixing this bug?

@IEvangelist @trrwilson is this resolved yet? Commenting to ensure this thread is still relevant.
